modules/Attendance/attendance.php

```diff
@@ -16,488 +16,448 @@
+                    AND gibbonAttendanceLogPerson.date LIKE :date
+                    AND gibbonAttendanceLogPerson.context = 'Roll Group'
+                    GROUP BY gibbonAttendanceLogRollGroup.gibbonAttendanceLogRollGroupID
+                    ORDER BY gibbonAttendanceLogPerson.timestampTaken";
+
+                    $resultLog=$connection2->prepare($sqlLog);
+                    $resultLog->execute($dataLog);
+                } catch (PDOException $e) {
+                    $page->addError($e->getMessage());
+                }
+
+                $log = $resultLog->fetch();
+
+                // general row variables
+                $row['rowID'] = 'gibbonRollGroupID';
+                $row['currentDate'] = Format::dateConvert($currentDate);
+
+                // render group link variables
+                $row['groupQuery'] = '/modules/Roll Groups/rollGroups_details.php';
+                $row['groupName'] = $row['name'];
+
+                // render recentHistory into the row
+                for ($i = count($lastNSchoolDays)-1; $i >= 0; --$i) {
+                    if ($i > (count($lastNSchoolDays) - 1)) {
+                        $dayData = [
+                            'currentDate' => null,
+                            'currentDayTimestamp' => null,
+                            'status' => 'na'
+                        ];
+                    } else {
+                        $dayData = [
+                            'currentDate' => Format::dateConvert($lastNSchoolDays[$i]),
+                            'currentDayTimestamp' => Format::timestamp($lastNSchoolDays[$i]),
+                            'status' => isset($logHistory[$lastNSchoolDays[$i]]) ? 'present' : 'absent',
+                        ];
+                    }
+                    $row['recentHistory'][] = $dayData;
+                }
+
+                // Attendance not taken
+                $row['today'] = ($resultLog->rowCount()<1) ? 'not taken' : 'taken';
+                $row['in'] = ($resultLog->rowCount()<1)? "" : ($log["total"] - $log["absent"]);
+                $row['out'] = $log["absent"];
+
+                $attendanceByRollGroup[] = $row;
+            }
+
+            // define DataTable
+            $attendanceByRollGroupTable = clone $dailyAttendanceTable;
+            $attendanceByRollGroupTable->setTitle(__('My Roll Group'));
+            if (isActionAccessible($guid, $connection2, "/modules/Attendance/attendance_take_byRollGroup.php")) {
+                $attendanceByRollGroupTable->addActionColumn()
+                    ->width('50px')
+                    ->setAction('takeAttendance')
+                    ->setLabel(__('Take Attendance'))
+                    ->setIcon('attendance')
+                    ->setURL(
+                        'index.php?' . http_build_query([
+                            'q' => '/modules/Attendance/attendance_take_byRollGroup.php',
```

Comment:
The index.php is already added by the Action class, so just `/modules/Attendance/attendance_take_byRollGroup.php` works for this method, then you can use `addParam(key, value)` or `addParams(array)` to add the url params.